yarn-project/archiver/src/archiver/memory_archiver_store/l1_to_l2_message_store.ts

```diff
@@ -46,93 +48,22 @@ export class NewL1ToL2MessageStore {
     }
     return messages;
   }
-}
 
-/**
- * A simple in-memory implementation of an L1 to L2 message store
- * that handles message duplication.
- */
-export class L1ToL2MessageStore {
   /**
-   * A map containing the entry key to the corresponding L1 to L2
-   * messages (and the number of times the message has been seen).
+   * Gets the L1 to L2 message index in the L1 to L2 message tree.
+   * @param l1ToL2Message - The L1 to L2 message.
+   * @returns The index of the L1 to L2 message in the L1 to L2 message tree.
    */
-  protected store: Map<bigint, L1ToL2MessageAndCount> = new Map();
-
-  constructor() {}
-
-  addMessage(entryKey: Fr, message: L1ToL2Message) {
-    const entryKeyBigInt = entryKey.toBigInt();
-    const msgAndCount = this.store.get(entryKeyBigInt);
-    if (msgAndCount) {
-      msgAndCount.count++;
-    } else {
-      this.store.set(entryKeyBigInt, { message, count: 1 });
-    }
-  }
-
-  getMessage(entryKey: Fr): L1ToL2Message | undefined {
-    return this.store.get(entryKey.value)?.message;
-  }
-
-  getMessageAndCount(entryKey: Fr): L1ToL2MessageAndCount | undefined {
-    return this.store.get(entryKey.value);
-  }
-}
-
-/**
- * Specifically for the store that will hold pending messages
- * for removing messages or fetching multiple messages.
- */
-export class PendingL1ToL2MessageStore extends L1ToL2MessageStore {
-  getEntryKeys(limit: number): Fr[] {
-    if (limit < 1) {
-      return [];
-    }
-    // fetch `limit` number of messages from the store with the highest fee.
-    // Note the store has multiple of the same message. So if a message has count 2, include both of them in the result:
-    const messages: Fr[] = [];
-    const sortedMessages = Array.from(this.store.values()).sort((a, b) => b.message.fee - a.message.fee);
-    for (const messageAndCount of sortedMessages) {
-      for (let i = 0; i < messageAndCount.count; i++) {
-        messages.push(messageAndCount.message.entryKey!);
-        if (messages.length === limit) {
-          return messages;
-        }
+  getMessageIndex(l1ToL2Message: Fr): bigint {
```

Comment:
We are still just getting the first here no? For #3976 we should be checking some nullifiers if we find multiple.

Reply:
Yes, will do the change in a separate PR.